xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/ListClass.java

```diff
@@ -974,10 +1013,36 @@ private option createOption(String value, String display, List<String> selectlis
      */
     public void fromList(BaseProperty<?> property, List<String> list)
     {
-        if (property instanceof ListProperty) {
-            ((ListProperty) property).setList(list);
+        fromList(property, list, false);
+    }
+
+    /**
+     * Set the passed {@link List} into the passed property.
+     *
+     * @param property the property to modify
+     * @param list the list to set
+     * @param filterEmptyValues if {@code true} filter out the empty values from the list.
+     * @since 14.2RC1
+     */
+    protected void fromList(BaseProperty<?> property, List<String> list, boolean filterEmptyValues)
```

Comment:
Maybe this new `filterEmptyValues` concept should be an actual configuration of the property class (that happen to be always true for group/user classes) instead of just a parameter in a protected method.

Reply:
Could be a global improvment to make indeed. Feels a bit too much for this PR though.